springfox-schema/src/main/java/springfox/documentation/schema/property/OptimizedModelPropertiesProvider.java

```diff
@@ -218,6 +220,49 @@ public boolean apply(Annotation input) {
 
   }
 
+  private boolean shouldBeIncludeDueToJsonView(BeanPropertyDefinition jacksonProperty,
+                                               ModelContext givenContext) {
+
+    Class classToRestrictOn = null;
+    JsonView jsonView = givenContext.getJsonView();
+    if (jsonView != null) {
+      classToRestrictOn = jsonView.value()[0];
+    }
+    List<Class> allInterfaces = getAllInterfaces(classToRestrictOn, new ArrayList<Class>());
+
+    if (allInterfaces.isEmpty()) {
+      return true;
+    } else {
+      Class[] views = jacksonProperty.findViews();
+      if (views != null) {
+        List<Class<?>> fieldViews = Arrays.asList(jacksonProperty.findViews());
+        for (Class eachView : fieldViews) {
+          if (allInterfaces.contains(eachView)) {
+            return true;
+          }
+        }
+      }
+    }
+
+    return false;
+  }
+
+  private static List<Class> getAllInterfaces(Class clazz, List<Class> collectedClasses) {
```

Comment:
Jackson support JsonView and herititance (cf https://spring.io/blog/2014/12/02/latest-jackson-integration-improvements-in-spring).
Maybe I need to extend this is we use class or interface. Right now Its works with interfaces